stl/inc/bit

```diff
@@ -211,18 +211,44 @@ _NODISCARD int _Checked_x86_x64_popcount(const _Ty _Val) noexcept {
 
 
 #if defined(_M_ARM) || defined(_M_ARM64)
+#ifdef __clang__ // workaround
+inline constexpr int _Clang_arm_arm64_countl_zero(const unsigned short _Val) {
+    return __builtin_clzs(_Val);
+}
+
+inline constexpr int _Clang_arm_arm64_countl_zero(const unsigned int _Val) {
+    return __builtin_clz(_Val);
+}
+
+inline constexpr int _Clang_arm_arm64_countl_zero(const unsigned long _Val) {
+    return __builtin_clzl(_Val);
+}
+
+inline constexpr int _Clang_arm_arm64_countl_zero(const unsigned long long _Val) {
+    return __builtin_clzll(_Val);
+}
+#endif // ^^^ __clang__ workaround ^^^
+
 template <class _Ty>
 _NODISCARD int _Checked_arm_arm64_countl_zero(const _Ty _Val) noexcept {
     constexpr int _Digits = numeric_limits<_Ty>::digits;
     if (_Val == 0) {
         return _Digits;
     }
 
+#ifdef __clang__ // workaround
+    if constexpr (sizeof(_Ty) < sizeof(unsigned short)) {
+        return _Clang_arm_arm64_countl_zero(static_cast<const unsigned short>(_Val));

```

Comment:
I believe that this is incorrect; `countl_zero` returns a different answer when the input value is widened by zero-filling. Example:
```
C:\Temp>type meow.cpp
```
```cpp
#include <bit>
#include <iostream>
using namespace std;

int main() {
  const unsigned char uc = 0b0000'0100;
  const unsigned short us = 0b0000'0000'0000'0100;
  static_assert(uc == us);
  cout << countl_zero(uc) << "\n";
  cout << countl_zero(us) << "\n";
}
```
```
C:\Temp>cl /EHsc /nologo /W4 /std:c++latest meow.cpp
meow.cpp

C:\Temp>meow
5
13
```

Reply:
Do you think I should just special case `unsigned char` or since I believe that is the only legal type smaller than short which doesn't have a clang builtin `countl`? Otherwise I can do the more general thing that libcxx does: https://github.com/llvm/llvm-project/blob/116177afcce88d807c1beffcb9221999ad8a69a9/libcxx/include/bit#L146-L176